### PR TITLE
Fix multiple LDAP configuration support by fixing AccessFactory

### LINKED ISSUE #37454: [Bug]: Multiple Ldap doesn't working
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hello,
I have two LDAP configuration s04 and s05.
I recently updated Nextcloud to 26.
But now when I try to connect, I can't. Ldap servers switch between them instead test on all LDAP configurations 

# php ./occ ldap:test-config s04           ---- I check status of my first LDAP
_The configuration is valid and the connection could be established!_

# php ./occ ldap:check-user toto     ---- I check user 
_The user is still available on LDAP._

# php ./occ ldap:check-user toto       ---- I check again with the same user (seems to test on my second config of LDAP...)
_The user does not exists on LDAP anymore.
Clean up the user's remnants by: ./occ user:delete "toto"_

# php ./occ ldap:test-config s04   ---- I do a check again my user work again
_The configuration is valid and the connection could be established!_
# php ./occ ldap:check-user toto
The user is still available on LDAP.

For now, I deleted my second LDAP's configuration and all seems to be ok except some users on my second LDAP who can't access to my nextcloud until I find a patch...

Thanks a lot

### Steps to reproduce

1. Create two LDAP's configuration
2. login with a user who appear only on one ldap
3.

### Expected behavior

User checked on all LDAP

### Installation method

None

### Nextcloud Server version

24

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
Enabled:
  - admin_audit: 1.16.0
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - contactsinteraction: 1.7.0
  - dashboard: 7.6.0
  - dav: 1.25.0
  - deck: 1.9.0
  - drawio: 2.1.0
  - encryption: 2.14.0
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_downloadactivity: 1.16.0
  - files_external: 1.18.0
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - firstrunwizard: 2.15.0
  - integration_openai: 1.0.2
  - keeweb: 0.6.12
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - nextcloud_announcements: 1.15.0
  - notes: 4.7.2
  - notifications: 2.14.0
  - oauth2: 1.14.0
  - password_policy: 1.16.0
  - photos: 2.2.0
  - picker: 1.0.4
  - privacy: 1.10.0
  - provisioning_api: 1.16.0
  - quota_warning: 1.16.0
  - recommendations: 1.5.0
  - related_resources: 1.1.0-alpha1
  - richdocuments: 8.0.0
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - spreed: 16.0.1
  - support: 1.9.0
  - survey_client: 1.14.0
  - suspicious_login: 4.4.0
  - systemtags: 1.16.0
  - tables: 0.3.3
  - tasks: 0.14.5
  - text: 3.7.2
  - theming: 2.1.1
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - user_ldap: 1.16.0
  - user_status: 1.6.0
  - user_usage_report: 1.10.0
  - viewer: 1.10.0
  - weather_status: 1.6.0
  - workflowengine: 2.8.0
Disabled:
  - accessibility: 1.10.0
  - activity: 2.18.0 (installed 2.13.2)
  - analytics: 4.8.0 (installed 4.8.0)
  - announcementcenter: 6.5.1 (installed 6.5.1)
  - bruteforcesettings: 2.6.0 (installed 2.4.0)
  - carnet: 0.24.6 (installed 0.24.6)
  - contacts: 5.2.0 (installed 5.2.0)
  - dashboardcharts: 0.1.3 (installed 0.1.3)
  - data_request: 1.13.0 (installed 1.13.0)
  - drop_account: 2.1.0 (installed 2.1.0)
  - duplicatefinder: 0.0.15 (installed 0.0.15)
  - end_to_end_encryption: 1.12.3 (installed 1.12.3)
  - epubreader: 1.4.7 (installed 1.4.7)
  - files_accesscontrol: 1.16.0 (installed 1.16.0)
  - files_automatedtagging: 1.16.1 (installed 1.16.1)
  - files_mindmap: 0.0.27 (installed 0.0.27)
  - files_retention: 1.15.0 (installed 1.15.0)
  - files_texteditor: 2.15.0 (installed 2.15.0)
  - files_videoplayer: 1.13.0
  - forms: 3.2.0 (installed 3.2.0)
  - groupfolders: 14.0.0 (installed 14.0.0)
  - mail: 3.0.2 (installed 3.0.2)
  - officeonline: 2.0.0 (installed 2.0.0)
  - ojsxc: 5.0.0 (installed 5.0.0)
  - quicknotes: 0.8.5 (installed 0.8.5)
  - terms_of_service: 2.1.0 (installed 2.1.0)
  - twofactor_totp: 8.0.0-alpha.0 (installed 6.4.1)
  - twofactor_u2f: 6.4.0-alpha.1 (installed 6.4.0-alpha.1)
  - user_external: 3.1.0 (installed 3.1.0)
  - user_saml: 5.1.2 (installed 5.1.2)
  - workflow_pdf_converter: 1.11.0 (installed 1.11.0)
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by come-nc**
* Resolves: #37454 

## Summary

PR #34772 broke the fix from PR #35070 because AccessFactory was reusing the same Manager instance for all access instances.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by blizzz — APPROVED** [review]

**Comment by evilham** [comment]
Tested this patch, it solves the issue on NC 26

**Review by szaimen — APPROVED** [review]
🐘

File: apps/user_ldap/lib/AccessFactory.php [hunk 1]
```diff
@@ -55,7 +53,7 @@ public function get(Connection $connection): Access {
 		return new Access(
 			$connection,
 			$this->ldap,
-			$this->userManager,
+			Server::get(Manager::class),
```
Comment (artonge) [inline apps/user_ldap/lib/AccessFactory.php:56]: Can you add a comment to explain why we do not use DI? It will help to not revert in the future.

**Review by artonge — APPROVED** [review]

**Pull request merged**